src/Drips.sol

```diff
@@ -359,34 +374,55 @@ abstract contract Drips {
+    /// @return historyHashes The history hashes valid for squeezing each of `dripsHistory` entries.
+    /// In other words history hashes which had been valid right before each drips
+    /// configuration was set, matching `dripsHistoryHash` emitted in its `DripsSet`.
+    /// The first item is always equal to `historyHash`.
+    /// @return currCycleConfigs The number of the sender's
+    /// drips configurations which have been seen in the current cycle.
+    /// This is also the number of used entries in each of the sender's `nextSqueezed` arrays.
     function _squeezeDripsResult(
         uint256 userId,
         uint256 assetId,
         uint256 senderId,
         bytes32 historyHash,
         DripsHistory[] memory dripsHistory
-    ) internal view returns (uint128 amt, uint256 squeezedNum, uint256[] memory squeezedIdxs) {
-        DripsState storage senderState = _dripsStorage().states[assetId][senderId];
-        _verifyDripsHistory(historyHash, dripsHistory, senderState.dripsHistoryHash);
+    )
+        internal
+        view
+        returns (
+            uint128 amt,
+            uint256 squeezedNum,
+            uint256[] memory squeezedRevIdxs,
+            bytes32[] memory historyHashes,
+            uint256 currCycleConfigs
+        )
+    {
+        {
+            DripsState storage sender = _dripsStorage().states[assetId][senderId];
+            historyHashes = _verifyDripsHistory(historyHash, dripsHistory, sender.dripsHistoryHash);
+            // If the last update was not in the current cycle,
+            // there's only the single latest history entry to squeeze in the current cycle.
+            currCycleConfigs = 1;
+            if (sender.updateTime >= _currCycleStart()) currCycleConfigs = sender.currCycleConfigs;
+        }
+        squeezedRevIdxs = new uint256[](dripsHistory.length);
         uint32[2 ** 32] storage nextSqueezed =
             _dripsStorage().states[assetId][userId].nextSqueezed[senderId];
         uint32 squeezeEnd = _currTimestamp();
-        // If the last update was not in the current cycle,
-        // there's only the single latest history entry to squeeze in the current cycle.
-        uint256 squeezedIdx = 1;
-        if (senderState.updateTime >= _currCycleStart()) squeezedIdx = senderState.currCycleConfigs;
-        uint256 historyIdx = dripsHistory.length;
-        squeezedIdxs = new uint256[](historyIdx);
-        while (squeezedIdx > 0 && historyIdx > 0) {
-            squeezedIdx--;
-            historyIdx--;
-            DripsHistory memory drips = dripsHistory[historyIdx];
+        for (uint256 i = 1; i <= dripsHistory.length && i <= currCycleConfigs; i++) {
+            DripsHistory memory drips = dripsHistory[dripsHistory.length - i];
             if (drips.receivers.length != 0) {
-                uint32 squeezeStart = nextSqueezed[squeezedIdx];
+                uint32 squeezeStart = nextSqueezed[currCycleConfigs - i];
```

Comment:
The name always confuses me a bit. Because the `squeezeStart` is only the squeeze start for the first squeeze in a new cycle in the other cases it is the `updateTime`.   It is kind of a `potentialsqueezeStart`